pkg/subctl/cmd/validate_deployment.go

```diff
@@ -17,46 +17,129 @@ package cmd
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/spf13/cobra"
-	"github.com/submariner-io/submariner-operator/pkg/internal/cli"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	smClientset "github.com/submariner-io/submariner/pkg/client/clientset/versioned"
+	"github.com/submariner-io/submariner/pkg/util"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/rest"
+
+	"github.com/submariner-io/submariner-operator/apis/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/pkg/internal/cli"
 )
 
 var validatePodsCmd = &cobra.Command{
-	Use:   "pods",
-	Short: "Validate the submariner pods",
-	Long:  "This command checks that all the submariner pods are running",
-	Run:   validatePods,
+	Use:   "deployment",
+	Short: "Validate the submariner deployment",
+	Long: "This command checks that the submariner components are properly deployed and running" +
+		" with no overlapping CIDRs.",
+	Run: validateSubmarinerDeployment,
 }
 
 func init() {
 	validateCmd.AddCommand(validatePodsCmd)
 }
 
-func validatePods(cmd *cobra.Command, args []string) {
+func validateSubmarinerDeployment(cmd *cobra.Command, args []string) {
 	configs, err := getMultipleRestConfigs(kubeConfig, kubeContext)
 	exitOnError("error getting REST config for cluster", err)
 
 	for _, item := range configs {
-		message := fmt.Sprintf("Validating submariner pods in %q", item.clusterName)
-		status.Start(message)
-		fmt.Println()
-		checkPods(item.config, OperatorNamespace)
+		submariner := getSubmarinerResource(item.config)
+		if submariner == nil {
+			status.QueueWarningMessage(submMissingMessage)
+			status.End(cli.Success)
+			return
+		}
```

Comment:
We need to start the status in order to end it if submariner is missing.

```suggestion
                status.Start(fmt.Sprintf("Retrieving Submariner resource from %q", item.clusterName))
                 
		submariner := getSubmarinerResource(item.config)
		if submariner == nil {
			status.QueueWarningMessage(submMissingMessage)
			status.End(cli.Success)
			continue
		}
		
		status.End(cli.Success)
```